gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/reactor/handler/AbstractAccessPointAcceptor.java

```diff
@@ -0,0 +1,102 @@
+import io.gravitee.common.event.EventManager;
+import io.gravitee.gateway.reactor.accesspoint.AccessPointEvent;
+import io.gravitee.gateway.reactor.accesspoint.ReactableAccessPoint;
+import jakarta.validation.constraints.NotNull;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * @author GraviteeSource Team
+ */
+public abstract class AbstractAccessPointAcceptor<T extends Acceptor<T>>
+    implements Acceptor<T>, EventListener<AccessPointEvent, ReactableAccessPoint> {
+
+    protected final ReactorHandler apiReactor;
+    private final String environmentId;
+    private final EventManager eventManager;
+    protected List<T> acceptors;
+
+    public AbstractAccessPointAcceptor(ReactorHandler apiReactor, final EventManager eventManager, final String environmentId) {
+        this.apiReactor = apiReactor;
+        this.environmentId = environmentId;
+        this.eventManager = eventManager;
+        this.eventManager.subscribeForEvents(this, AccessPointEvent.class);
+    }
+
+    protected void initializeAcceptors(final List<ReactableAccessPoint> initialReactableAccessPoints) {
+        this.acceptors =
+            initialReactableAccessPoints
+                .stream()
+                .filter(reactableAccessPoint -> reactableAccessPoint.getTarget() == getTarget())
+                .map((ReactableAccessPoint reactableAccessPoint) -> createAcceptor(reactableAccessPoint.getHost()))
+                .collect(toCollection(CopyOnWriteArrayList::new));
+    }
+
+    protected abstract T createAcceptor(String accessPointHost);
+
+    protected abstract ReactableAccessPoint.Target getTarget();
+
+    public abstract String toString();
+
+    @Override
+    public void clear() {
+        this.eventManager.unsubscribeForEvents(this, AccessPointEvent.class);
+        if (this.acceptors != null) {
+            this.acceptors.forEach(Acceptor::clear);
+        }
+    }
+
+    @Override
+    public void onEvent(Event<AccessPointEvent, ReactableAccessPoint> event) {
+        if (!environmentId.equals(event.content().getEnvironmentId())) {
+            return;
+        }
+        if (event.content().getTarget() != getTarget()) {
+            return;
+        }
+
+        T acceptor = createAcceptor(event.content().getHost());
+        AccessPointEvent type = event.type();
+
+        if (Objects.requireNonNull(type) == AccessPointEvent.DEPLOY) {
+            if (acceptors == null) {
+                acceptors = new CopyOnWriteArrayList<>();
+            }
+            acceptors.add(acceptor);
+        } else if (type == AccessPointEvent.UNDEPLOY && acceptors != null) {
+            acceptors.remove(acceptor);
+        }
+    }
+
+    @Override
+    public ReactorHandler reactor() {
+        return apiReactor;
+    }
+
+    @Override
+    public int compareTo(@NotNull T o) {
+        return this.toString().toLowerCase().compareTo(o.toString().toLowerCase());
```

Comment:
Maybe we can leave it to the child implementation instead and remove the need for the toString()?